OPHD/Technology/TechnologyReader.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "Technology.h"
+
+#include <map>
+#include <string>
+
+class TechnologyReader
+{
+public:
+	TechnologyReader() = delete;
+	TechnologyReader(const std::string& techFile);
+
+	std::map<int, Technology> technologies;

```

Comment:
See that's what I was thinking -- I originally called it TechnologyTree but it didn't feel quite right since it's not stored or handled as a tree. So I though perhaps TechnologyCatalog (to match other naming conventions) or TechnologyLibrary.

Reply:
I think "tech tree" is pretty standard terminology. Besides, it's not really about how it's stored. It's more about the graph of dependencies between techs, which forms a tree. In particular, there are no cycles in the dependency graph, which is an important part of what makes the graph a tree.

[Wikipedia: Tree (graph theory)](https://en.wikipedia.org/wiki/Tree_(graph_theory))
> In graph theory, a tree is an undirected graph in which any two vertices are connected by exactly one path, or equivalently a connected acyclic undirected graph. A forest is an undirected graph in which any two vertices are connected by at most one path, or equivalently an acyclic undirected graph, or equivalently a disjoint union of trees.

I guess you could call it a `TechForest`, just in case not all techs are connected by a dependency path. :wink: